js/viz/series/base_series.js

```diff
@@ -485,7 +485,35 @@ Series.prototype = {
         }
     },
 
-    _drawElements: function(animationEnabled, firstDrawing, translateAllPoints) {
+    _preparePositionSegments() {
+        const that = this;
+        const points = that._points || [];
+        const closeSegment = points[0] && points[0].hasValue() && that._options.closed;
+
+        const segments = points.reduce(function(segments, p) {
+            const segment = segments[segments.length - 1];
+
+            if(!p.translated) {
+                p.setDefaultCoords();
+            }
+
+            if(p.hasValue() && p.hasCoords()) {
+                segment.push(p);
+            } else if(!p.hasValue()) {
```

Comment:
Don't use `&&` instead of `if` plz
```suggestion
            } else if(!p.hasValue() && segment.length) {
                segments.push([]);
            }
```

Reply:
fixed